File: Exemplos_Internet/WindowsForms/# Editor_Componetes/GridThemes_src/GridThemesDemoVB/App_Code/ColorUtility.vb
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports System.Drawing

Public Class ColorUtility

    Public Shared Function BlendColors(ByVal color1 As Color, ByVal color2 As Color) As Color
        ' return the average between both colors



        Dim r As Integer = (CType(color1.R, Integer) + CType(color2.R, Integer)) / 2
        Dim b As Integer = (CType(color1.B, Integer) + CType(color2.B, Integer)) / 2
        Dim g As Integer = (CType(color1.G, Integer) + CType(color2.G, Integer)) / 2

        Return System.Drawing.Color.FromArgb(r, g, b)
        
    End Function


    Public Shared Function BlendColors(ByVal color1 As Color, ByVal sColor2 As String) As Color
        ' return the average between both colors; the second is supplied as a string
        Dim color2 As Color = System.Drawing.ColorTranslator.FromHtml(sColor2)
        Return ColorUtility.BlendColors(color1, color2)

    End Function


End Class

File: Exemplos_Internet/WindowsForms/# Editor_Componetes/GridThemes_src/GridThemesDemoCS/App_Code/ColorUtility.cs
using System;
using System.Data;
using System.Configuration;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using System.Drawing;

/// <summary>
/// Summary description for ColorUtility
/// </summary>
public class ColorUtility
{
    public static Color BlendColors(Color color1, Color color2)
    {
        // return the average between both colors
        int r = (color1.R + color2.R) / 2;
        int b = (color1.B + color2.B) / 2;
        int g = (color1.G + color2.G) / 2;

        return System.Drawing.Color.FromArgb(r, g, b);
    }

    public static Color BlendColors(Color color1, string sColor2)
    {
        // return the average between both colors; the second is supplied as a string
        Color color2 = System.Drawing.ColorTranslator.FromHtml(sColor2);
        return BlendColors(color1, color2);
    }
}
